### PR TITLE
Add FirtoolBinaryPath option

### PR DESCRIPTION
**Pull request by GeorgeLyon**
In certain environments, such as when running tests via `bloop`, it may be difficult to influence which `firtool` CIRCT uses via the PATH environment variable (the only method currently available). This allows us to replace the version of `firtool` that is used at runtime (or point Chisel to the correct firtool binary in scenarios where it isn't on your PATH).

This will also be helpful if we want to package firtool in the Chisel jar @jackkoenig, since we will probably be extracting it to a temporary directory and not in the PATH.

#### Type of Improvement

- Feature (or new API)

#### Release Notes

- Add `FirtoolBinaryPathOption` to select a different `firtool` binary at runtime.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.5.x`, `3.6.x`, or `5.x` depending on impact, API modification or big change: `6.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: src/main/scala/circt/stage/Annotations.scala [hunk 1]
```diff
@@ -101,6 +101,8 @@ case class EmittedMLIR(
 
 }
 
+case class FirtoolBinaryPath(option: String) extends NoTargetAnnotation with CIRCTOption
```
Comment (jackkoenig) [inline src/main/scala/circt/stage/Annotations.scala:104]: This should probably have a ScalaDoc as well as a shell option form

Comment (GeorgeLyon) [reply src/main/scala/circt/stage/Annotations.scala:105]: I added a scaladoc... couldn't quite figure out how to add a shell option but if you are invoking from a shell you can always do `PATH=path/to/firtool:$PATH <command>` so not sure if it is 100% necessary.

**Comment by GeorgeLyon** [comment]
@jackkoenig I just updated this to add a `HasShellOptions` case object for the FirtoolBinaryPath annotation. I'm not sure I did this right as I'm not sure how to test, and it seems `HasShellOptions` uses that registeredlibrary magic and isn't actually connected to the annotation case class, but declared on a separate object that then creates the appropriate annotation.

Is this correct?

File: src/main/scala/circt/stage/Annotations.scala [hunk 2]
```diff
@@ -101,6 +101,20 @@ case class EmittedMLIR(
 
 }
 
+case object FirtoolBinaryPath extends HasShellOptions {
```
Comment (seldridge) [inline src/main/scala/circt/stage/Annotations.scala:104]: Nit:

```suggestion
private[stage] case object FirtoolBinaryPath extends HasShellOptions {
```

**Review by jackkoenig — APPROVED** [review]

**Pull request merged**